server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -107,15 +108,33 @@ public final void collectExistingBucket(LeafBucketCollector subCollector, int do
      * Refer to that method for documentation about the merge map.
      */
     public final void mergeBuckets(long[] mergeMap, long newNumBuckets) {
+        UnaryOperator<Long> mergeMapOperator = new UnaryOperator<Long>() {
+            @Override
+            public Long apply(Long bucket) {
+                return mergeMap[Math.toIntExact(bucket)];
+            }
+        };
+
+        mergeBuckets(mergeMapOperator, newNumBuckets);
+    }
+
+    /**
+     * This only tidies up doc counts. Call {@link MergingBucketsDeferringCollector#mergeBuckets(UnaryOperator)} to
+     * merge the actual ordinals and doc ID deltas.
+     */
+    public final void mergeBuckets(UnaryOperator<Long> mergeMap, long newNumBuckets){
```

Comment:
Two things, only one is important though:
1. Could you replace `UnaryOperator<Long>` with `LongUnaryOperator`? That'll prevent auto-boxing of the parameter.
2. Could you switch the order of the arguments? I think the call sites are a little prettier when the "function" argument is last, if possible.